src/Telemetry/Telemetry.cs

```diff
@@ -122,6 +122,23 @@ private void InitializeTelemetry(string productVersion, IConfiguration config)
             }
         }
 
+        public void TrackSQLClientEvent(string eventName, IDictionary<string, string> properties = null)
```

Comment:
This shouldn't go in the telemetry class at all. Just have the event listener handle doing the logging.